byway/src/main/webapp/track-interests.js

```diff
@@ -0,0 +1,55 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+let selected = [];
+
+/** 
+ * Track the selected interests in an array according to their status. 
+ * @param elem tracks the current button element chosen.
+ */
+function checkStatus(elem) {
+  if(elem.className === "btnActive") {
```

Comment:
Note that elements can have multiple classes, so checking equality like this may be error-prone. CSS classes should be used for styling not for program logic.

In general it would be a good idea to try to separate your *data* from how that data is *presented* to the user; Storing data in JS and then just using HTML to present that data makes the data easier to reason about, because it has a known location and a consistent structure, and it makes the page easier to reason about and keep properly in sync, because the page is nothing more than a way of presenting a specific portion of the data.

In this case it may be better to track things like selected buttons in code and just "render" the results into the page.

Reply:
I wasn't entirely sure how to handle the active status of buttons so resorted to doing it this way. The issue with rendering it as they click is I'd have to keep track of their location to keep the same button on the screen (I assume). Users could also click buttons out of order, but if I rendered it per array element there's no guarantee the elements would be in the same spot nor in the same order. That's why I went with rendering and making a visual change (color change) to indicate to the users which are selected at a time.